skimage/measure/block.py

```diff
@@ -62,6 +62,9 @@ def block_reduce(image, block_size, func=np.sum, cval=0, **func_kwargs):
         raise ValueError("`block_size` must have the same length "
                          "as `image.shape`.")
 
+    if func_kwargs is None:
```

Comment:
Why not just have func_kwargs={} as the default argument in line 5?

Reply:
It's considered bad form to put in a mutable data structure as the default argument to a function, because it is instantiated only once. If someone changes the code so that the mutable is changed (e.g. `func_kwargs.update(...)`), it gets modified for every subsequent call to that function. See e.g. [here](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) and [here](https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/), but Google provides a vast treasure trove of people feeling the pain here.